Sources/SwifterSwift/SwiftStdlib/StringExtensions.swift

```diff
@@ -978,6 +990,45 @@ public extension String {
     }
     #endif
 
+    #if canImport(Foundation)
+    /// SwifterSwift: Overload Swift's 'contains' operator for matching regex
+    ///
+    /// - Parameter lhs: String to check on regex.
+    /// - Parameter rhs: Regex to match against.
+    /// - Returns: `true` if there is at least one match for the regex in the string.
+    static func ~= (lhs: String, rhs: NSRegularExpression) -> Bool {
+        let range = NSRange(lhs.startIndex..<lhs.endIndex, in: lhs)
+        return rhs.firstMatch(in: lhs, range: range) != nil
+    }
+    #endif
+
+    #if canImport(Foundation)
+    /// SwifterSwift: Returns a new string in which all occurrences of a regex pattern in a specified range of the receiver are replaced by the template.
+    /// - Parameter pattern: Regex pattern to replace.
+    /// - Parameter template: The regex template to replace the pattern.
+    /// - Parameter searchRange: The range in the receiver in which to search.
+    /// - Returns: A new string in which all occurrences of regex pattern in searchRange of the receiver are replaced by template.
+    func replacingOccurrences(of pattern: String, with template: String,
```

Comment:
1. This can be generalised to `StringProtocol`.
2. Users should be able to pass `CompareOptions`. I've added that `assert` because the documentation says that only those options can be used with it.

```Swift
extension StringProtocol {
    #if canImport(Foundation)
    func replacingOccurrences<Target, Replacement>(
        ofPattern pattern: Target,
        withTemplate template: Replacement,
        options: String.CompareOptions = [.regularExpression],
        range searchRange: Range<Self.Index>? = nil) -> String where Target: StringProtocol, Replacement: StringProtocol {
        assert(options.isStrictSubset(of: [.regularExpression, .anchored, .caseInsensitive]), "Invalid options for regular expression replacement")
        return replacingOccurrences(of: pattern, with: template, options: options.union(.regularExpression), range: searchRange)
    }
    #endif
}
```

Reply:
Fine by me, thanks